inbc-program/inbc/parser/source_app_parser.py

```diff
@@ -35,8 +35,8 @@ def application_add(args: argparse.Namespace) -> str:
 
     manifest += '<repo><repos>'
 
-    for source in args.sources:
-        manifest += '<source_pkg>' + source + '</source_pkg>'
+    source_tags = [f'<source_pkg>{source}</source_pkg>' for source in args.sources]
+    manifest += ''.join(source_tags)
 
     manifest += ('</repos>'
                  f'{create_xml_tag(arguments, "filename")}</repo>'
```

Comment:
**Suggestion:** Use a more efficient way to concatenate strings for XML generation, such as using a list and joining it only once. [performance]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    parts = ['<repo><repos>']
    parts.extend(f'<source_pkg>{source}</source_pkg>' for source in args.sources)
    parts.append('</repos>')
    parts.append(create_xml_tag(arguments, "filename"))
    parts.append('</repo>')
    manifest += ''.join(parts)
```